youtube_dl/extractor/cozytv.py

```diff
@@ -0,0 +1,114 @@
+                'channel_id': 'althype',
+                'channel_url': 'https://cozy.tv/althype',
+                'duration': 7349,
+                'view_count': 804,
+                'webpage_url': 'https://cozy.tv/althype/replays/2022-04-22',
+            }
+        },
+        {
+            'url': 'https://cozy.tv/beardson/replays/2022-05-01_2',
+            'md5': '2f474ea9404fe90ca87a6390a6e10173',
+            'info_dict': {
+                'id': '626eb2b73f2f1a99756b6a1b',
+                'title': '24 HOUR STREAM',
+                'ext': 'mp4',
+                'display_id': '2022-05-01_2',
+                'thumbnail': r're:^https?://.*/replays/beardson/2022-05-01_2/thumb\.webp$',
+                'uploader': 'beardson',
+                'creator': 'beardson',
+                'release_timestamp': 1651421852,
+                'release_date': '20220501',
+                'timestamp': 1651421852,
+                'upload_date': '20220501',
+                'uploader_id': 'beardson',
+                'uploader_url': 'https://cozy.tv/beardson',
+                'channel': 'beardson',
+                'channel_id': 'beardson',
+                'channel_url': 'https://cozy.tv/beardson',
+                'duration': 313,
+                'view_count': 656,
+                'webpage_url': 'https://cozy.tv/beardson/replays/2022-05-01_2',
+            }
+        }
+    ]
+
+    def _real_extract(self, url):
+        video_id, user = re.match(self._VALID_URL, url).group('id', 'user')
+        data = self._download_json('https://api.cozy.tv/cache/%s/replay/%s' % (user, video_id), video_id)
+
+        cdn_url = '/'.join((data['cdns'][0], 'replays', user, video_id)).rstrip('/')
+        user_url = 'https://cozy.tv/' + user
+
+        info = {
+            'id': data.get('_id') or video_id,
+            'title': data['title'],
+        }
+
+        timestamp = unified_timestamp(data.get('date'))
+        formats = []
+        formats.extend(self._extract_m3u8_formats(
+            cdn_url + '/index.m3u8',
+            video_id,
+            ext='mp4',
+            m3u8_id='hls'))
+        formats.extend(self._extract_m3u8_formats(
+            cdn_url + '/index-ts-audio.m3u8',
+            video_id,
+            ext='mp4',
+            m3u8_id='hls-audio',
+            fatal=False))
+        self._sort_formats(formats)
+
+        info.update({
+            'channel': user,
+            'channel_id': user,
+            'channel_url': user_url,
+            'creator': user,
+            'display_id': video_id,
+            'duration': float_or_none(data.get('duration')),
+            'formats': formats,
+            'release_timestamp': timestamp,
+            'thumbnail': cdn_url + '/thumb.webp',
+            'timestamp': timestamp,
+            'uploader': user,
+            'uploader_id': user,
+            'uploader_url': user_url,
+            'view_count': int_or_none(data.get('peakViewers')),
+            'webpage_url': url,
+        })
```

Comment:
Finally, let's check whether the metadata makes sense.

* is `user` actually the creator on the site, or just the uploader?
* according to the definition (for historical reasons yt-dl metadata is based on what YouTube offers) `uploader` is the user's actual name while `uploader_id` is the user's ID on the site: does it make sense to extract the same value to both?
* similarly for `channel`/`channel_id`?
* does the site have "channels" that justify extracting `channel*` with the same values as `uploader*` (considering what the IE doc says)?
 > ... channel fields may or may not repeat uploader fields. This depends on a particular extractor.
* should `release_timestamp` be set if it just repeats `timestamp`?